code/modules/admin/verbs/one_click_antag.dm

```diff
@@ -35,35 +35,41 @@ client/proc/one_click_antag()
 	var/list/mob/living/carbon/human/candidates = list()
 	var/mob/living/carbon/human/H = null
 
-	log_admin("[key_name(owner)] tried making Traitors with One-Click-Antag")
-	message_admins("[key_name_admin(owner)] tried making Traitors with One-Click-Antag")
-
-	for(var/mob/living/carbon/human/applicant in player_list)
-		if(ROLE_TRAITOR in applicant.client.prefs.be_special)
-			if(player_old_enough_antag(applicant.client,ROLE_TRAITOR))
-				if(!applicant.stat)
-					if(applicant.mind)
-						if(!applicant.mind.special_role)
-							if(!jobban_isbanned(applicant, "traitor") && !jobban_isbanned(applicant, "Syndicate"))
-								if(!(applicant.mind.assigned_role in temp.restricted_jobs))
-									if(!(applicant.client.prefs.species in temp.protected_species))
-										candidates += applicant
-
-	if(candidates.len)
-		var/numTratiors = min(candidates.len, 3)
-
-		for(var/i = 0, i<numTratiors, i++)
-			H = pick(candidates)
-			H.mind.make_Tratior()
-			candidates.Remove(H)
-
-		return 1
-
-
+	var/antnum = input(owner, "How many traitors you want to create? (0 - 5)","Amount:", 0) as num
+	var/confirmed = alert(owner, "Are you sure?", "Confirm creation", "Yes", "No")
+
+	if(!antnum)
+		return
+	if(antnum > 5 || antnum <= 0)

```

Comment:
Merge the two `if`s into `!antnum || antnum > 5 || antnum <= 0`. Byond is a lazy evaluation language.